### ISSUE TITLE
hydraJobs function support in nix flake check/nix flake show

### ISSUE THREAD

**Issue by shlevy** [NixOS/nix#7031]
Hydra currently supports a `hydraJobs` that is a function taking inputs like release.nix can be. This may be accidental due to implementation details, but it's actually a feature which is needed to do actual continuous **integration** (e.g. overriding a dependency in `flake.lock` with the latest revision). `nix flake check` and `nix flake show` should not error in this case, and ideally they'd autocall to actually check the individual jobs if all function args are defaulted.

**Comment by edolstra** [comment]
That's a bug in Hydra. For hermetic, reproducible evaluation, `hydraJobs` should not accept any arguments.

**Comment by shlevy** [comment]
I disagree that it's a bug, will move the discussion to NixOS/hydra#1247 until that's resolved.

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
